pkg/datastore/db_user.go

```diff
@@ -36,6 +36,41 @@ type dbUserSpec struct {
 	newRowsCond      string   // SQL conditional expression to be checked for rows being inserted or updated. Only those rows for which the condition is true will be inserted/updated
 }
 
+func (m dbUserSpec) MarshalJSON() ([]byte, error) {
+	data := struct {
```

Comment:
nit: non-blocking, Do you need a new struct to marshal, cant you use dbUserSpec itself? is it just for casing?

Reply:
I do need a new [anonymous] struct. "encoding/json" package won't add private fields to JSON string it's generating. One way to go around this is, every time a struct gets converted to JSON, to generate a new [anonymous] struct with all fields public, and convert that struct to JSON.
@krishnamiriyala